### PR TITLE
Provides Body Components for Automobile

### PR DESCRIPTION
**Pull request by Brandonhenk**
# Purpose 
Customizing luxary, economy, and sport Cabin, Engine Compartment, Quarterpanels, and Trunk Compartment.
body componets redid - page 1

### CODE REVIEW

**Comment by anthonykulis** [comment]
make sure to group these in to domain driven categories. If I am building a 2 door sports car with a luxury interior, I shouldn't have to understand all the nuances of your modeling. Right now as the diagram shows, its pretty anemic. Make it stand out. It makes sense I should be able to import `...body.sports.twodoor.LuxuryCabin` or something similar in concept. Its on you guys to meet my goals for a ubiquitous domain language. There are three other teams that all have to meet that as well. This way when we start phase two, we can use these models from phase 1 quite simply. If we keep these anemic models as you have described, the team using them is going to have to confer with you when they shouldn't need to.

**Comment by Brandonhenk** [comment]
body 1 - page 1-5

@anthonykulis  I know there would be more fields and methods in all of this, but as for BodyComponents is this the idea of the fields correct? Then for luxury, eco, sport would extend BodyComponets.

**Comment by anthonykulis** [comment]
i am totally confused by that to be quite honest. stop worrying about what component goes were for the moment. this about the domain language you want people to use. `body -> luxury -> fourdoorluxurybody` or maybe `body -> sport -> twodoorsport` or even `body -> suv -> 4doorhatch`. Design that concept *first* and it will be obvious were you put the parts like "trunk" in.

**Comment by Brandonhenk** [comment]
body 2 - page 1-3

**Comment by anthonykulis** [comment]
Guys, I am not too much liking this idea anymore. I was thinking that 2door, 4door, 5door as a top tier is too opaque. To me, and I want to know what you think, if I was to build an automobile, I would build a `car` or a `truck`. Lets say I was to build a `car`, I would build a `sedan`, a `coupe`, a `hatchback`, etc. From there maybe i have a `sportsedan` or a `sportcoupe`. All of these have distinct body styles, do they not? Same with truck, if I choose a truck, I'm going to get a bed vs a trunk, but I can then choose an `extendedcab`, a `crewcab`, a `standardcab` truck, etc. 

I think that might clearer. Dunno tho.

**Comment by Brandonhenk** [comment]
bodycomponents - page 1
body 2 - page 1-4

**Comment by anthonykulis** [comment]
that looks pretty good. some things are oddly added, which I will discuss with your team in person. I also need to see for each property their visibility and types. Once you have that checked off, then go add the methods parts. Once that is done, you can slice this up and everyone is one the same page because they will know exactly what to put in to the classes they write.

**Comment by anthonykulis** [comment]
let me get the latest and greatest UML please. you can submit them as you get them done.

**Comment by Brandonhenk** [comment]
bodycomponents - page 1-2
body 2 - page 1-5

**Comment by anthonykulis** [comment]
Ok. gonna have a bunch of replies coming.

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 14 41 am

With this. An economy van is not a SUV. A SportJeep is not a SUV. Because you set a property `hatchback`, now all jeeps have a hatch back. Jeep fans will be like NOPE. A van definitely doesn't have a hatch back. Ever. If you are just trying to fill space, put them in the right spot.

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 14 36 am

This is not UML. I want data types and visibility. Anyone should be able to pick up your UML and code it the way you want it. The way you have it lets me interpret how I want to code it.

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 14 24 am

Reword. There is no RegularTruck, especially when we are talking cabins. `StandardCab` is what I think you are looking for.

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 13 52 am

You do not need to force fit stuff. An engine compartment will have a hood. No doubt. That should be a property. If you want a variation of that for sport as you kind of show, they you will want an `EngineCompartment` and have the two variations `Standard` and `Sport`

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 13 36 am

`Seats`? Use `Seat`.

**Comment by anthonykulis** [comment]
screen shot 2017-03-10 at 10 13 04 am

`EconomyCabin` seems like it is named wonky, yes? maybe use `StandardCabin`

**Comment by anthonykulis** [comment]
Here are things you *have* to fix. 

1. You have spelling errors in nearly 100% of the items listed. Use spell check.
2. You need to clearly define what a property is. `+ Cabin cabin`. With that I know its `public Cabin cabin;` Get all those properties written into UML. This goes for accessors/mutators and `toString`

**Comment by Brandonhenk** [comment]
I went through and fixed the errors and tried to fill it all out with methods and ect. 
bodycomponents - page 1-4
bodycomponents - page 1-5
body 2 - page 1-6

**Comment by anthonykulis** [comment]
Uml looks much better. Typically we will put :returntype with a method even
if obvious. Come next week we will introduce interfaces and this makes much
more sense there

**Comment by Brandonhenk** [comment]
Here is my section of UML i have to code.
body 2 - page 1-9
body 2 - page 1-10
bodycomponents - page 1-8

**Comment by Brandonhenk** [comment]
Does this make sense with the getCabin I have for the SUV part? The SUV would override it. How would i make this work for the truck? @anthonykulis 
body 2 - page 1-11

**Comment by anthonykulis** [comment]
You mean with the three getCabin? If SUV getCabin doesn't matter, you can
make it abstract as well

**Comment by Brandonhenk** [comment]
Yeah thats what I was talking about. I understand now but for the truck I'm confused. For example would CrewCab have +getCabin() : LuxuryCabin? How do we know if its luxury or standard cabin? Do I get to pick sense I'm in designer?

**Comment by anthonykulis** [comment]
Yeah. You should be able to abstract getCabin(): Cabin and then implement
as getCabin(): LuxuryCabin.

**Comment by Brandonhenk** [comment]
Update on UML that I will code.

body 2 - page 1-19
body 2 - page 1-17
bodycomponents - page 1-10

**Comment by Brandonhenk** [comment]
@anthonykulis  what do i do if my code won't compile because my team mates are coding parts that my code has to have to run and they haven't made tests yet so i can't merge in their work? Now i can't make tests for mine because it won't compile.

**Comment by anthonykulis** [comment]
Ok, you are saying compile time errors, not infinite loop right? I pulled your repo and yeah, you have a ton of issues. First, your import statements are incorrect. I fixed that, but then you try to instantiate classes not in your repository. It seems you blocked yourself while you are waiting on everyone else. You're gonna have to get their stuff done, tested, etc, merge it in, then finish yours. Hopefully you didn't slice up your work so you have a circular dependency.

**Comment by anthonykulis** [comment]
oh wait. i missed that stuff. give me a sec to look more

File: edu/jalc/automobile/onlinebuilder/components/body/TestEconomySuv.java [hunk 1]
```diff
@@ -1,5 +1,17 @@
 package edu.jalc.automobile.onlinebuilder.components.body;
+import edu.jalc.automobile.parts;
```
Comment (anthonykulis) [inline edu/jalc/automobile/onlinebuilder/components/body/TestEconomySuv.java]: not a syntactically correct import statement. you're trying to import a directory not its contents

File: edu/jalc/automobile/onlinebuilder/components/body/TestEconomySuv.java [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
 
 public class TestEconomySuv{
-  
+  public void testGetCabin(){
+    System.out.println("Testing getCabin() :: EconomySuv");
+    StandardCabin standardCabin = new StandardCabin();
```
Comment (anthonykulis) [inline edu/jalc/automobile/onlinebuilder/components/body/TestEconomySuv.java]: Your constructor for this requires a parameter.

**Comment by anthonykulis** [comment]
Im doing some comments as code review, and some here. So pay attention. Some things I am not going to dig for because your commit messages are opaque to the task you created.

**Comment by anthonykulis** [comment]
You have `QuarterPanels` everywhere when you only have `QuarterPanel`
screen shot 2017-03-23 at 1 10 13 pm

**Comment by anthonykulis** [comment]
You are trying to use `StandardCabin` but never imported it. 

https://github.com/Brandonhenk/automobile/blob/master/edu/jalc/automobile/onlinebuilder/components/body/EconomySuv.java#L8

**Comment by anthonykulis** [comment]
You are trying to use `StandardTrunk` which doesn't exist. Nor should it. I dont have a standard trunk, I have a `Trunk`.
screen shot 2017-03-23 at 1 14 11 pm

**Comment by anthonykulis** [comment]
Ok. Dude. I can't do this. You just have tons of user introduced bugs. Its 99% attention to detail stuff. "Cant find symbol" is most of them. That simply means you either have a typo, didn't import, or class doesn't exist. Compile this in the terminal and scroll up. Fix the first one, compile, fix, compile, fix.

**Comment by Brandonhenk** [comment]
Updated UML for Brandon
bodycomponents - page 1-5
bodycomponents - page 1-6
body 2 - page 1-6
body 2 - page 1-7

**Pull request merged**